packages/shared/src/utils.ts

```diff
@@ -182,3 +184,11 @@ export function getDims(dataArray: NdArray): Dims {
   const [rows, cols] = dataArray.shape;
   return { rows, cols };
 }
+
+export function isValid(value: number, invalidValue?: number) {
+  if (invalidValue === undefined) {
+    return true;
+  }
+
+  return Math.abs((value - invalidValue) / invalidValue) > 1e-2;
```

Comment:
Usual problem when comparing floats, the equality check with `===` may not be satisfied due to a lack of precision.

So instead, I consider the value invalid if it lies in `+/- 1%` of `invalidValue`. This criteria is arbitrary and may be revisited in the future.

Reply:
Why is this an issue?
`number` is supposed to be float64, and only `_FillValue` should be considered invalid, right?

Reply:
My bad, netCDF specs defines the comparison to do:

 > If the _FillValue is positive then it defines a valid maximum, otherwise it defines a valid minimum. For integer types, there should be a difference of 1 between the _FillValue and this valid minimum or maximum. For floating point types, the difference should be twice the minimum possible (1 in the least significant bit) to allow for rounding error.

Reply:
I'm surprised that this is needed as well. With binary transfer, I don't get how we can lose precision. Perhaps there is a problem in the example file, or does this only happen with mock data?

There are lots of [edge cases](https://floating-point-gui.de/errors/comparison/) to comparing floats, unfortunately... [This package](https://github.com/micksatana/nearly-equal/blob/master/src/nearly-equal.ts) implements the `nearlyEqual` function from the article, but it's no longer maintained (though we could duplicate the function).

[This package](https://www.npmjs.com/package/float-equal) seems promising as well, as it is 1) making use of `Number.EPSILON`; 2) avoiding divisions; 3) adjusting the epsilon as a last resort: https://github.com/sindresorhus/float-equal/blob/main/index.js

If we go down the float comparison route, we definitely need to invoke the comparison code only when necessary to avoid computing something like `Number.EPSILON * Math.min(Math.abs(a), Math.abs(b))` on integers, for instance... 😅 

**EDIT**: Disregard the above since NetCDF specifies a comparison algorithm. It is all the more important that the comparison function be provided by the code that knows about NetCDF4, i.e. `@h5web/app`.

Reply:
After checking, it seems the loss of precision happens **in the binary transfer**.

By using JSON to get data, I get `1000000000000000` for the values in the array as expected.

But with the binary transfer, I get `999999986991104` instead :shrug:

Reply:
It is due to the fact that the data is stored in `float32` that is not able to store `1e15` in an accurate way. 

By using the JSON transfer, we use JS `Number` that is `float64` and therefore able to store `1e15` accurately.